### ISSUE TITLE
Different sound output between console and GUI application

### ISSUE THREAD

**Issue by tekrongames** [jarikomppa/soloud#181]
Hi, I'm getting weird and different results with sound files when setting it as a gui application (_-mwindows_) However, everything seems to be normal and I got the expected results in a console app. Some outputs are repeating sounds in gui setting. Any ideas?

Specs: Windows 7 x64, MinGW StaticLib Win32, latest stable release

**Comment by jarikomppa** [comment]
First step, the latest stable is rather old. It's better to use the tip of
the tree from git. I'm trying to get a new stable release out sometime this
year, but the head is already in a much better shape than the previous
release. (documentation is naturally lagging)

Other than that, maybe the back end changes for some reason (if the SDL
dlls are available or not, for example), or maybe you've set the buffer
size so small that it causes buffer overruns?

**Comment by tekrongames** [comment]
@jarikomppa Thanks, but still getting the same errors only on window mode (_-mwindows_) even with the latest commit here. Great, I would like to try that, but how do I set buffer size with Soloud, can you provide a quick example?

**Comment by jarikomppa** [comment]
It's one of the parameters of the init function. You can query the buffer
size that was automatically selected (with getBackendBufferSize) for a
value to start with. The default is typically a safe value (i.e, bigger
than absolutely necessary), but ymmv.

**Comment by tekrongames** [comment]
Thanks. I see and it seems to be working now with a few changes on my side. One question and somehow related to this issue, can I check if a particular sample is currently playing or assign channels and manage them? probably something like _sample.playing()_

**Comment by jarikomppa** [comment]
There's nothing exactly like that, but I'd like to hear more about your use
case to decide on the implementation.

**Comment by tekrongames** [comment]
I see. ok sounds good, thanks. Something like this with OpenAL

https://www.gamedev.net/forums/topic/518057-openal----retrieve-truefalse-if-source-is-playing/

**Comment by jarikomppa** [comment]
No, I don't want to know how this is implemented in some other audio API, I
want to know why you need the feature.

**Comment by tekrongames** [comment]
I know and not the intention, ok so with the current api, how do you play a sound source once and check if it stops playing before looping or playing it again? Say a trigger of a repeating event (keydown, input or condition) that plays a sound sample from a source, how do you make sure it completes before playing it again from the same source?

feature wise, actually not just me, but pretty much a necessity and standard.. 

https://docs.unity3d.com/ScriptReference/AudioSource-isPlaying.html
https://urho3d.github.io/documentation/1.7/class_urho3_d_1_1_sound_source.html#aa2a9cb9efed89b2973361f1275c36c75

In short, how do you know or check if a sound is currently playing or being played from a sound source or sample? 
https://social.msdn.microsoft.com/Forums/sqlserver/en-US/b9ab40bc-2432-434d-91c1-62b036f6af8e/detecting-if-sound-is-being-played?forum=Vsexpressvb

I am new to Soloud so I don't know how to achieve this with the current version of the library, so appreciate any suggestions or how-tos.

Thanks.

**Comment by jarikomppa** [comment]
Hi,

SoLoud can play a single source several times at the same time. When you
ask it to play, you get a handle to the instance (which you can discard if
you don't need it). The handle can be used to query whether it's valid or
not, if it's no longer valid, the sound is not in use.

If you want a sound to loop, just turn the looping on on the sound source
and the sound will loop. If you only want a single instance of a sound
playing, either call stop on the sound source before playing it again, or
mark it as single instance.

There's no general "is this sound source in use" query at the moment, but
one could be relatively easy to add. The reason why I'm asking about your
use case is that there are several ways to do so, some take more memory and
some take more processing power.

As a workaround you could just keep track of all the handles you get from
playing that sound source and check if any of them are valid..

**Comment by tekrongames** [comment]
I see, can you send the specific function to manage and check those handles? How do you mark it as a single instance?

Btw, closing this issue now. I have confirmed that the echoing problem is a bit complicated and in case you have some ideas as to why it has different output or if you can also do some tests between windows and console app, it would help a lot.

Thanks.

**Comment by jarikomppa** [comment]
http://soloud-audio.com/ has documentation.
